packages/core/data/mm/world/moon.yml

```diff
@@ -120,22 +120,21 @@
 "Moon Trial Link Iron Knuckle Fight":
   region: MOON
   exits:
-    "Moon Trial Link Rest 2": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs)"
-    "Moon Trial Link Rest 3": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs) && has_bombchu && has_arrows"
+    "Moon Trial Link Rest 2": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs || has_bombchu || (has(MASK_BLAST) && masks(5)))"
+    "Moon Trial Link Rest 3": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs || has_bombchu || (has(MASK_BLAST) && masks(5))) && ((has_bombchu && has_arrows) || short_hook_anywhere)"
   locations:
-    "Moon Trial Link Iron Knuckle Chest": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs)"
-  gossip:
-    "Moon Trial Link Gossip Left": "has_bombchu && has_arrows"
-    "Moon Trial Link Gossip Right": "has_bombchu && has_arrows"
+    "Moon Trial Link Iron Knuckle Chest": "soul_iron_knuckle && (has_weapon || has_mask_goron || has_bombs || has_bombchu || (has(MASK_BLAST) && masks(5)))"
 "Moon Trial Link Rest 3":
   region: MOON
   exits:
     "Moon Trial Link Iron Knuckle Fight": "true"
-    "Moon Trial Link Exit": "has_bombchu && can_use_fire_arrows"
+    "Moon Trial Link Exit": "(has_bombchu || (short_hook_anywhere && has(MASK_BLAST) && masks(5))) && can_use_fire_arrows"
   locations:
     "Moon Trial Link HP": "true"
   gossip:
     "Moon Trial Link Gossip Last": "true"
+    "Moon Trial Link Gossip Left": "true"
+    "Moon Trial Link Gossip Right": "true"
```

Comment:
You can access these `Gossip Left` and `Gossip Right` without fighting the iron knuckle, with short hookshot anywhere